test/AutoDiff/TBD/derivative_symbols.swift

```diff
@@ -0,0 +1,56 @@
+// RUN: %target-swift-frontend -emit-ir -o/dev/null -parse-as-library -module-name test -validate-tbd-against-ir=all %s
+// RUN: %target-swift-frontend -emit-ir -o/dev/null -parse-as-library -module-name test -validate-tbd-against-ir=all %s -O
+// RUN: %target-swift-frontend -emit-ir -o/dev/null -parse-as-library -module-name test -validate-tbd-against-ir=missing %s -enable-testing
+// RUN: %target-swift-frontend -emit-ir -o/dev/null -parse-as-library -module-name test -validate-tbd-against-ir=missing %s -enable-testing -O
+
```

Comment:
I removed some lines from the test:
```swift
// RUN: %empty-directory(%t)
// RUN: %target-swift-frontend -typecheck -parse-as-library -module-name test %s -emit-tbd -emit-tbd-path %t/typecheck.tbd
// RUN: %target-swift-frontend -emit-ir -parse-as-library -module-name test %s -emit-tbd -emit-tbd-path %t/emit-ir.tbd
// RUN: diff -u %t/typecheck.tbd %t/emit-ir.tbd
```

These seem to crash on Linux: `swift::writeTBDFile` calls MachO APIs, which seems Darwin-specific.
Some but not all TBD tests use `REQUIRES: VENDOR=apple`.
<details>
<pre>
unexpected platform
UNREACHABLE executed at llvm-project/llvm/lib/TextAPI/MachO/TextStubCommon.cpp:60!
Stack dump:
0.      Program arguments: build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift -frontend -target x86_64-unknown-linux-gnu -module-cache-path build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_
64/swift-test-results/x86_64-unknown-linux-gnu/clang-module-cache -swift-version 4 -ignore-module-source-info -typo-correction-limit 10 -typecheck -parse-as-library -module-name test swift/test/AutoDiff/TBD/derivative_symbols.swift -emit-tbd -emit
-tbd-path build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/test-linux-x86_64/AutoDiff/TBD/Output/derivative_symbols.swift.tmp/typecheck.tbd 
1.      Swift version 5.3-dev (LLVM 898aa0b6a4, Swift acd7816780)
 #0 0x0000000005112474 PrintStackTraceSignalHandler(void*) (build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift+0x5112474)
 #1 0x000000000511006e llvm::sys::RunSignalHandlers() (build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift+0x511006e)
 #2 0x000000000511274c SignalHandler(int) (build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift+0x511274c)
 #3 0x00007fd0807f1890 __restore_rt (/lib/x86_64-linux-gnu/libpthread.so.0+0x12890)
 #4 0x00007fd07ee5ce97 raise /build/glibc-OTsEL5/glibc-2.27/signal/../sysdeps/unix/sysv/linux/raise.c:51:0
 #5 0x00007fd07ee5e801 abort /build/glibc-OTsEL5/glibc-2.27/stdlib/abort.c:81:0
 #6 0x00000000050685b6 (build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift+0x50685b6)
 #7 0x0000000004c7c625 llvm::yaml::ScalarTraits<llvm::SmallSet<llvm::MachO::PlatformKind, 3u, std::less<llvm::MachO::PlatformKind> >, void>::output(llvm::SmallSet<llvm::MachO::PlatformKind, 3u, std::less<llvm::MachO::PlatformKind> > const&, void*, llvm::raw_ostream&) (/home/dan
ielzheng/swift-master/build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift+0x4c7c625)
 #8 0x0000000004c7b1f2 std::enable_if<has_ScalarTraits<llvm::SmallSet<llvm::MachO::PlatformKind, 3u, std::less<llvm::MachO::PlatformKind> > >::value, void>::type llvm::yaml::yamlize<llvm::SmallSet<llvm::MachO::PlatformKind, 3u, std::less<llvm::MachO::PlatformKind> > >(llvm::yam
l::IO&, llvm::SmallSet<llvm::MachO::PlatformKind, 3u, std::less<llvm::MachO::PlatformKind> >&, bool, llvm::yaml::EmptyContext&) (build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift+0x4c7b1f2)
 #9 0x0000000004c6c798 llvm::yaml::MappingTraits<llvm::MachO::InterfaceFile const*>::mapKeysToValues(llvm::MachO::FileType, llvm::yaml::IO&, llvm::MachO::InterfaceFile const*&) (build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift
+0x4c6c798)
#10 0x0000000004c6b0e2 llvm::MachO::TextAPIWriter::writeToStream(llvm::raw_ostream&, llvm::MachO::InterfaceFile const&) (build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift+0x4c6b0e2)
#11 0x00000000007a147b swift::writeTBDFile(swift::ModuleDecl*, llvm::raw_ostream&, swift::TBDGenOptions const&) (build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift+0x7a147b)
#12 0x00000000005723da swift::writeTBD(swift::ModuleDecl*, llvm::StringRef, swift::TBDGenOptions const&) (build/Ninja-ReleaseAssert+stdlib-Release/swift-linux-x86_64/bin/swift+0x5723da)
</pre>
<details>